src/main/java/mate/academy/springboot/datajpa/controller/ProductController.java

```diff
@@ -0,0 +1,89 @@
+package mate.academy.springboot.datajpa.controller;
+
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.AllArgsConstructor;
+import mate.academy.springboot.datajpa.dto.ProductRequestDto;
+import mate.academy.springboot.datajpa.dto.ProductResponseDto;
+import mate.academy.springboot.datajpa.dto.mapper.ProductMapper;
+import mate.academy.springboot.datajpa.model.Category;
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.service.CategoryService;
+import mate.academy.springboot.datajpa.service.ProductService;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@AllArgsConstructor
+@RequestMapping("/product")
+public class ProductController {
+    private final ProductService productService;
+    private final ProductMapper productMapper;
+    private final CategoryService categoryService;
+
+    @GetMapping("/inject")
+    public void saveProducts() {
+        Product product = new Product();
+        product.setTitle("ad");
+        product.setPrice(BigDecimal.valueOf(1349));
+        Category category = Category.builder().name("1").build();
+        categoryService.save(category);
+        product.setCategory(category);
+        productService.save(product);
+        productService.findAllByPriceBetween(BigDecimal.valueOf(1200), BigDecimal.valueOf(1500)).forEach(System.out::println);
+    }
+
+    @PostMapping
+    public ProductResponseDto create(@RequestBody ProductRequestDto requestDto) {
+        Product product = productService.save(productMapper.toModel(requestDto));
+        return productMapper.toResponseDto(product);
+    }
+
+    @GetMapping("/{id}")
+    public ProductResponseDto getById(@PathVariable Long id) {
+        Product product = productService.getById(id);
+        return productMapper.toResponseDto(product);
+    }
+
+    @PutMapping
+    public ProductResponseDto update(@RequestBody ProductRequestDto requestDto) {
```

Comment:
create a separate dto for update request. How you can update a product without its id?